test/wait-task/pkg/reconciler/reconciler.go

```diff
@@ -67,27 +73,56 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, r *v1alpha1.Run) kreconc
 		return nil
 	}
 
-	dur, err := time.ParseDuration(expr.Value.StringVal)
-	if err != nil {
-		r.Status.MarkRunFailed("InvalidDuration", "The duration param was invalid: %v", err)
+	// Skip if the Run is cancelled.
+	if r.IsCancelled() {
+		logger.Infof("The Custom Task Run %v has been cancelled", r.GetName())
+		r.Status.CompletionTime = &metav1.Time{Time: c.Clock.Now()}
+		r.Status.MarkRunFailed("Cancelled", "The Wait Task is cancelled")
```

Comment:
```suggestion
                var reason string
                if r.spec.StatusMessage == "" {
                    reason = "Cancelled"
                } else {
                    reason = r.Spec.StatusMessage
                }
		r.Status.MarkRunFailed(reason, "The Wait Task is cancelled")
```
although I'm not sure if `StatusMessage` should be the "reason" or the "message"-- maybe check what the taskrun reconciler does

Reply:
https://github.com/tektoncd/pipeline/blob/af4d0e6eb018a34c950180939a7f6a2f24941da6/pkg/apis/pipeline/v1beta1/taskrun_types.go#L100-L106

The `StatusMessage` is message, do you think distinguishing the different cancellation is a better idea?

Reply:
Ah OK in that case, I would do `r.Status.MarkRunFailed("Cancelled", r.spec.StatusMessage)` (if status message != "")-- does that seem reasonable?

Reply:
SG! Thanks for catching this~ updated!